src/test/java/com/exactpro/th2/dataservice/healer/HealerTest.java

```diff
@@ -47,42 +49,47 @@
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.mockito.Mockito.when;
 
 @RunWith(JUnit4.class)
 public class HealerTest {
-    /**
-     * This rule manages automatic graceful shutdown for the registered server at the end of test.
-     */
-    @Rule
-    public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
-
-    private final HealerConfiguration configuration = new HealerConfiguration("test", "1", 100);
-    private final CrawlerInfo crawlerInfo = CrawlerInfo.newBuilder().setId(CrawlerId.newBuilder().setName("testCrawler")).build();
-    private final CradleStorage storageMock = mock(CradleStorage.class);
-    private final List<StoredTestEventWrapper> events = new ArrayList<>();
-
-
-    @SuppressWarnings("unchecked")
-    private final StreamObserver<DataServiceInfo> dataServiceResponseObserver
-            = (StreamObserver<DataServiceInfo>) mock(StreamObserver.class);
-
-    @SuppressWarnings("unchecked")
-    private final StreamObserver<EventResponse> eventResponseObserver
-            = (StreamObserver<EventResponse>) mock(StreamObserver.class);
-
-    private HealerServiceImpl healer;
+    private static final String HEALER_NAME = "healer";
+    private static final String HEALER_VERSION = "1";
+    private static final String CRAWLER_NAME = "crawler";
+    private static final String PARENT_EVENT_ID = "parent_event_id";
+    private static final String CHILD_EVENT_ID = "child_event_id";
+    private static final String GRANDCHILD_EVENT_ID = "grandchild_event_id";
+    private static final HealerConfiguration CONFIGURATION = new HealerConfiguration(HEALER_NAME, HEALER_VERSION, 100);
+    private static final CrawlerId CRAWLER_ID = CrawlerId.newBuilder().setName(CRAWLER_NAME).build();
+    private static final CrawlerInfo CRAWLER_INFO = CrawlerInfo.newBuilder().setId(CRAWLER_ID).build();
+    private static final CradleStorage STORAGE_MOCK = mock(CradleStorage.class);
+    private static final List<StoredTestEventWrapper> events = new ArrayList<>();
+
+    private static Server server;
+    private static ManagedChannel channel;
+    private static DataProcessorGrpc.DataProcessorBlockingStub blockingStub;
 
     @BeforeEach
-    public void prepare() throws Exception {
-        healer = new HealerServiceImpl(configuration, storageMock);
+    public void prepare() throws IOException, CradleStorageException {
+        int port = 8081;
+
+        server = ServerBuilder.forPort(port)
+                .addService(new HealerImpl(CONFIGURATION, STORAGE_MOCK))
+                .build()
+                .start();
+        channel = ManagedChannelBuilder.forAddress("localhost", port)
+                .usePlaintext()
+                .directExecutor()
+                .build();
```

Comment:
Please, take a look at this example here
https://github.com/grpc/grpc-java/blob/master/examples/src/test/java/io/grpc/examples/routeguide/RouteGuideServerTest.java

We should use InProcessServerBuilder and InProcessChannelBuilder